internal/exec/util/file.go

```diff
@@ -169,6 +170,27 @@ func (u Util) SetPermissions(mode *int, node types.Node) error {
 	return nil
 }
 
+// toFileMode converts Go permission bits to POSIX permission bits.
+func toFileMode(m *int) os.FileMode {
```

Comment:
The only caller already knows that the mode argument is non-nil, so let's simplify by dereferencing the pointer in the caller and taking an integer argument here.